vm/src/stdlib/os.rs

```diff
@@ -221,6 +221,37 @@ fn convert_nix_errno(vm: &VirtualMachine, errno: Errno) -> PyClassRef {
     }
 }
 
+// Flags for os_access
+const F_OK: u8 = 0;
+const R_OK: u8 = 4;
+const W_OK: u8 = 2;
+const X_OK: u8 = 1;
+
+// TODO : Implement W_OK and X_OK
+fn os_access(path: PyStringRef, mode: u8, vm: &VirtualMachine) -> PyResult<bool> {
+    let path = path.as_str();
+    let file_metadata = fs::metadata(path);
+    match mode {
+        F_OK => match file_metadata {
+            Ok(_) => Ok(true),
+            Err(err) => {
+                if err.kind() == ErrorKind::NotFound {
+                    Ok(false)
+                } else {
+                    Err(convert_io_error(vm, err))
+                }
+            }
+        },
+        R_OK => match file_metadata {
+            Ok(metadata) => Ok(metadata.permissions().readonly()),
+            Err(err) => Err(convert_io_error(vm, err)),
+        },
```

Comment:
`readonly()` checks to see if it has read but *not* write permissions, which isn't what `R_OK` signals. You should probably try to parse the mode of the file using the [`PermissionsExt`](https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html) unix-only trait and [conditional compilation](https://doc.rust-lang.org/rust-by-example/attribute/cfg.html)